docs/sources/operations/query-fairness/_index.md

```diff
@@ -0,0 +1,121 @@
+fairness across these actors within the tenants. To achieve that, Loki
+introduced hierarchical scheduler queues in version 2.9 based on
+[LID 0003: Query fairness across users within tenants]({{<relref "../../lids/0003-QueryFairnessInScheduler.md">}})
+and they are enabled by default.
+
+## What are hierarchical queues and how do they work
+
+To understand hierarchical queues, we first need to know that in the scheduler
+component each tenant has it's own FIFO queue where (sub-)queries are enqueued.
+Depending on whether shuffle sharding is turned off or on, either all queriers
+or a subset of those queriers are allowed to dequeue from a tenant queue.
+
+Tenant queues are the first level of the queue hierarchy. When a tenant
+executes a query without any further controls, all of its sub-queries are
+enqueued to the first level queue.
+
+However, tenant queues can also have sub-queues. When a task (sub-request) is
+dequeued, the dequeuing algorithm of the queue picks a task round robin across
+the first level (local queue) and all its sub-queues.
+
+![Hierarchical queues](./hierarchical-queues.png)
+
+The figure above shows that a tenant queue has a local queue, which is a leaf
+node in the queue tree, and a set of sub-queues. Each sub-queue, again like the
+tenant queue, consists of a local queue, and possible sub-queues, resulting in
+a recursive tree structure.
+
+So, how can we make use of these tree-like queue structure to achieve query fairness?
+
+## How to control query fairness
+
+As already mentioned, by default, sub-queries are only enqueued at the first
+(tenant) level of the queue tree. The tenant is provided by the `X-Scope-OrgID`
+header that is required when running Loki in multi-tenant mode.
+
+The HTTP header `X-Loki-Actor-Path` can be used to control to which sub-queue a
+query (or more correctly its sub-queries) is enqueued.
+
+The following example shows a `curl` command that invokes the HTTP endpoint for range queries
+and passes both the `X-Scope-OrgID` and the `X-Loki-Actor-Path` headers.
+
+```bash
+curl -s http://localhost:3100/loki/api/v1/query_range?xxx \
+    -H 'X-Scope-OrgID: grafana' \
+    -H 'X-Loki-Actor-Path: joe'
+```
+
+The query that this request invokes ends up in the sub-queue `joe` of the
+tenant queue `grafana`. Another user can use their own name in the actor path
+header to enqueue their queries in their own sub-queue.
+
+Since the tenant queue chooses the next task in a round-robin manner, both
+actors (in our case human users) get their 50% share when a querier dequeues
+from that tenant.
+
+Even when there are tasks in the local queue of the tenant, the local queue get
+1/3 and each sub-queue gets 1/3 of their share.
+
+With N actors, each actor get 1/Nth of their share.
+
+Since explained implementation and the header name already suggest, it is
+possible to enqueue queries several levels deep. To do so, you can construct a
```

Comment:
"it is possible to enqueie queries several levels deep"  Since we don't recommend going more than one or two levels deep, do we even want to let people know this is possible?  Because you know someone is going to try it.  Could they cause problems for themselves?